Second commit, Pushing all apps files to Develop

Termoss opened the pull request.

Termoss committed:
Second commit, Pushing all apps files to Develop

mateusz-dziurdziak commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+/.idea/
+/target/
+/termos-bookstore.iml
```
Idealnie jest żeby wszystkie pliki *.iml były excludowane. Ktoś importując Wasz projekt może nazwać go inaczej

mateusz-dziurdziak commented on pom.xml:
```diff
@@ -0,0 +1,117 @@
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.1.2.RELEASE</version>
```
Są już nowsze wersje Spring Boot (taki tam komentarz)

mateusz-dziurdziak commented on pom.xml:
```diff
@@ -0,0 +1,117 @@
+
+    <!-- Java 11 -->
+    <properties>
+        <java.version>1.11</java.version>
```
Zmieniła sie numeracja. Powinno tutaj być `11`, a nie `1.11`

mateusz-dziurdziak commented on src/main/java/com/termos/StartBookApplication.java:
```diff
@@ -0,0 +1,16 @@
+        SpringApplication.run(StartBookApplication.class, args);
+    }
+
+    /* run this only on profile 'demo', avoid run this in test */
```
Kometarz niepotrzebny

mateusz-dziurdziak commented on src/main/java/com/termos/controller/OrderController.java:
```diff
@@ -0,0 +1,95 @@
+        if(orderService.getById(id) == null){
+            response.isSuccess = false;
+            response.message = "Order doesn't exist.";
+            return ResponseEntity.badRequest().body(response);
```
W przypadku gdy coś nie istnieje powinniście zwracać notFound

mateusz-dziurdziak commented on src/main/java/com/termos/controller/UserController.java:
```diff
@@ -0,0 +1,101 @@
+            response.isSuccess = false;
+            return ResponseEntity.badRequest().body(response);
+        }
+        u.authorities = user.authorities;
```
W javie raczej stosuje się zapisy z getterami i setterami

mateusz-dziurdziak commented on src/main/java/com/termos/dto/BookDTO.java:
```diff
@@ -0,0 +1,31 @@
+
+    @NotNull
+    @NotEmpty
+    public float price = 0.00f;
```
NotNull i NotEmpty nie ma tutaj sensu ponieważ float jest typem prostym (nie jest to obiekt). Taki typ nie może być pusty i zawsze ma jakąś wartość

mateusz-dziurdziak commented on src/main/java/com/termos/dto/OrderDTO.java:
```diff
@@ -0,0 +1,103 @@
+import java.sql.Timestamp;
+
+
+public class OrderDTO extends AbstractDTO{
```
Zastosujcie lomobk

mateusz-dziurdziak commented on src/main/java/com/termos/repository/AbstractRepository.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.List;
+
+@Component
+public interface AbstractRepository<T extends AbstractModel> {
```
To jest interfejs tylko - adnotacja Component tutaj nie ma sensu

mateusz-dziurdziak commented on src/main/java/com/termos/repository/AuthorityRepository.java:
```diff
@@ -0,0 +1,116 @@
+                list.add(auth);
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```
Wyjątki powinny być logowane za pomocą `log.errro`

mateusz-dziurdziak reviewed (commented):
Jest naprawdę spoko - nie mam jakiś większych uwag.

Trochę mało testów + powinniście mocniej wyczyścić kod. Jak macie lombok to nie musicie tworzyć tyle getterów/setterów/konstruktorów i metod toString

Termoss committed:
Modyfied  README.md,
Changed Version Spring Boot in pom.xml.
Changed Java Version in pom.xml.
Deleted  notFound and added  badRequest in Controllers,
Deleted Not null and not empty annotation in BookDTO,
Deleted @Component annotation in AbstractRepository,
Add Slf4j annotation and change e.printStackTrace(); to log.error.
Add Lombok annotation in Order DTO
Added more tests UsersControlerTest and OrdersControlerTest
